src/Microsoft.ML.Transforms/Text/LdaTransform.cs

```diff
@@ -930,46 +657,475 @@ public void Dispose()
             }
         }
 
-        private ColumnType[] InitColumnTypes(int numTopics)
+        private sealed class Mapper : MapperBase
         {
-            Host.Assert(Utils.Size(Infos) > 0);
-            var types = new ColumnType[Infos.Length];
-            for (int c = 0; c < Infos.Length; c++)
-                types[c] = new VectorType(NumberType.Float, numTopics);
-            return types;
+            private readonly LdaTransformer _parent;
+            private readonly ColumnType[] _srcTypes;
+            private readonly int[] _srcCols;
+
+            public Mapper(LdaTransformer parent, Schema inputSchema)
+                : base(parent.Host.Register(nameof(Mapper)), parent, inputSchema)
+            {
+                _parent = parent;
+                _srcTypes = new ColumnType[_parent.ColumnPairs.Length];
+                _srcCols = new int[_parent.ColumnPairs.Length];
+
+                for (int i = 0; i < _parent.ColumnPairs.Length; i++)
+                {
+                    if (!inputSchema.TryGetColumnIndex(_parent.ColumnPairs[i].input, out _srcCols[i]))
+                        throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", _parent.ColumnPairs[i].input);
+
+                    var srcCol = inputSchema[_srcCols[i]];
+                    if (!srcCol.Type.IsKnownSizeVector || !(srcCol.Type.ItemType is NumberType))
+                        throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", _parent.ColumnPairs[i].input);
+
+                    _srcTypes[i] = srcCol.Type;
+                }
+            }
+
+            public override Schema.Column[] GetOutputColumns()
+            {
+                var result = new Schema.Column[_parent.ColumnPairs.Length];
+                for (int i = 0; i < _parent.ColumnPairs.Length; i++)
+                {
+                    var info = _parent._columns[i];
+                    result[i] = new Schema.Column(_parent.ColumnPairs[i].output, new VectorType(NumberType.Float, info.NumTopic), null);
```

Comment:
>_parent.ColumnPairs[i].output [](start = 50, length = 29)

is this equal to `info.Input`?

Reply:
this is equal to info.Output.  If output column name is not specified then info.Output is same as info.Input.

Is there any concern here ?

---
In reply to: [233584111](https://github.com/dotnet/machinelearning/pull/1410#discussion_r233584111) [](ancestors = 233584111)

Reply:
well, `info.Output` is shorter, so I'd rather use it.

---
In reply to: [233740244](https://github.com/dotnet/machinelearning/pull/1410#discussion_r233740244) [](ancestors = 233740244,233584111)

Reply:
keeping this as-is

We are using SaveColumns from OneToOneTransformer base.  So when loading up models,  we should be using  _parent.ColumnPairs[i].output   (info.Output  may be null in those cases)

---
In reply to: [234857190](https://github.com/dotnet/machinelearning/pull/1410#discussion_r234857190) [](ancestors = 234857190,233740244,233584111)